### DISCUSSION TITLE
[Frontend] [Questions about UI] React + FC + Typescript + Redux + Hooks + Immer

### DISCUSSION THREAD

**Discussion by Allianzcortex** [realworld-apps/realworld#682] [General]
Recently I submitted my implementations of real-world into the list : 

Github repo :  https://github.com/Allianzcortex/react-typescript-hooks-realworld
Demo link : https://react-typescript-hooks-realworld.vercel.app/
Tech stack : React Functional Component + Immer + Redux + Hooks + axios + Testing Library

In the past I have pushed 2 PRs for the flask-realworld backend(https://github.com/gothinkster/flask-realworld-example-app/pulls?q=is%3Apr+author%3AAllianzcortex) , so I  really take this project seriously. However I just know that it got rejected because of UI reasons. 

here is the UI :
![image](https://user-images.githubusercontent.com/7290780/121882043-67be4800-cce6-11eb-8934-3f7c10d0f7ce.png)
here is loader to show the progress : 
![image](https://user-images.githubusercontent.com/7290780/121882065-6e4cbf80-cce6-11eb-91aa-89450a538c9d.png)here is snack bar to show the notification : 
![image](https://user-images.githubusercontent.com/7290780/121882088-7573cd80-cce6-11eb-9559-2d1147dddb0f.png)

This is other implementation（e.g. https://angular.realworld.io/) : 
![image](https://user-images.githubusercontent.com/7290780/121882118-7d337200-cce6-11eb-8860-eba1ecb47d60.png)

So I'm really a little confused here. When we talk about the UI , is it required that we must use bootstrap and apply all the theme (I guess not because I can see the difference between https://react-mobx.realworld.io/#/ and https://angular.realworld.io/). So could I get more detailed feedback about it so I can make sure I'm on the right way.

**Comment by geromegrignon** [comment]
Hello @Allianzcortex,

The UI requirements are fully part of the specs.
However thanks for pointing out the difference : the current Angular implementation uses the url of another repository.
I'll create an issue on the related repo.

About UI enhancements, we'll soon start working on a new version of the specs.
I'll keep you in touch if you want to contribute to provide a more modern user experience.

**Reply by Allianzcortex** [reply]
If we don't have a mobile spec then I feel it should also not be mandatory for frontend design( I don't like the word of `unfortunately`  [here](https://github.com/gothinkster/realworld/tree/master/spec#stylestemplates-1)  it's a good thing for developers to make full use of their design talents). Default bootstrap4 theme is great but we should also allow other themes(e.g. semantic, material. Now we have `ant design for react` , `vuetify for vue`  which is more commonly used than bootstrap). It will be a great thing if ***FREE STYLE*** can be encouraged here.

**Reply by Allianzcortex** [reply]
see the related [discussion](https://github.com/gothinkster/realworld/issues/662#issuecomment-846619549) . So I guess I will submit my version again after V2 is released ( - :

**Reply by geromegrignon** [reply]
The reason is already included into the specs : 

> We created a custom Bootstrap 4 style & templates to **ensure all frontends had consistent UI functionality**. Our starter kit includes all the templates & info required to get up and running..

**Reply by Allianzcortex** [reply]
It's very useful in V1 and also glad to see the benefits of diversity in V2 ！ 🎉🎉 consistent functionality with different UI styles  🍺🍺

**Reply by geromegrignon** [reply]
Keep in mind that we are considering allowing ui libraries/frameworks diversity (not free style). But we'll open RFC (Request for Comment) discussions to move to the next version of the specs soon. I'll keep you in touch if you want to participate.
